lib/integrations/awsoidc/listdatabases_iam_config.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/gravitational/trace"
+
+	awslib "github.com/gravitational/teleport/lib/cloud/aws"
+)
+
+const (
+	// defaultPolicyNameForListDatabases is the default name for the Inline Policy added to the IntegrationRole.
+	defaultPolicyNameForListDatabases = "ListDatabases"
+)
+
+// ConfigureIAMListDatabasesRequest is a request to configure the required Policy to use the List Databases action.
+type ConfigureIAMListDatabasesRequest struct {
+	// Region is the AWS Region.
+	// Used to set up the AWS SDK Client.
+	Region string
+
+	// IntegrationRole is the Integration's AWS Role used by the integration.
+	IntegrationRole string
+
+	// ListDatabasesPolicy is the Policy Name that is created to allow access to call AWS APIs.
+	// Defaults to ListDatabases
+	ListDatabasesPolicy string
+}
+
+// CheckAndSetDefaults ensures the required fields are present.
+func (r *ConfigureIAMListDatabasesRequest) CheckAndSetDefaults() error {
+	if r.Region == "" {
+		return trace.BadParameter("region is required")
+	}
+
+	if r.IntegrationRole == "" {
+		return trace.BadParameter("integration role is required")
+	}
+
+	if r.ListDatabasesPolicy == "" {
+		r.ListDatabasesPolicy = defaultPolicyNameForListDatabases
+	}
+
+	return nil
+}
+
+// ListDatabasesIAMConfigureClient describes the required methods to create the IAM Policies required for Listing Databases.
+type ListDatabasesIAMConfigureClient interface {
+	// PutRolePolicy creates or replaces a Policy by its name in a IAM Role.
+	PutRolePolicy(ctx context.Context, params *iam.PutRolePolicyInput, optFns ...func(*iam.Options)) (*iam.PutRolePolicyOutput, error)
+}
+
+// NewListDatabasesIAMConfigureClient creates a new ListDatabasesIAMConfigureClient.
+func NewListDatabasesIAMConfigureClient(ctx context.Context, region string) (ListDatabasesIAMConfigureClient, error) {
+	if region == "" {
+		return nil, trace.BadParameter("region is required")
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(region))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return iam.NewFromConfig(cfg), nil
+}
```

Comment:
nit: I would just remove this function altogether and put it inline in `teleport.go`. Or at least return `*iam.Client` instead of an interface

Reply:
Done :+1: